Check autoconfs $cross_compiling instead as AC_CANONICAL_HOST call
will fail if the host cpu is not recognised (which can happen when
e.g. Yocto builds for "allarch").

Signed-off-by: Jussi Kukkonen <jussi.kukkonen@intel.com>
Upstream-Status: Backport [cc276dfa41]

diff --git a/configure.ac b/configure.ac
index 5b48b1a..3d45a4b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -15,13 +15,10 @@ AC_CONFIG_MACRO_DIR([m4])
 
 AC_SUBST([WAYLAND_PROTOCOLS_VERSION], [wayland_protocols_version])
 
-AC_CANONICAL_HOST
-AC_CANONICAL_BUILD
-
 AC_ARG_VAR([wayland_scanner], [The wayland-scanner executable])
 AC_PATH_PROG([wayland_scanner], [wayland-scanner])
 if test x$wayland_scanner = x; then
-        if test x$host = x$build; then
+        if test "x$cross_compiling" != "xyes"; then
                 PKG_CHECK_MODULES(WAYLAND_SCANNER, [wayland-scanner])
                 wayland_scanner=`$PKG_CONFIG --variable=wayland_scanner wayland-scanner`
         else
-- 
cgit v0.10.2

